### DISCUSSION TITLE
Setting up Code for i and python

### DISCUSSION THREAD

**Discussion by a4ginatl** [codefori/vscode-ibmi#1964] [Q&A] [answered]
I don't know where to ask this question to get help so am asking it here. Please pint me in a different direction if I need to go elsewhere.

I use Code for i for all my development work.
I am starting to learn to use python on the IBM i.
I installed the latest update of python 3 on the system.
I installed python editors in VSC.

Problem: I am not able to get VSC, python to work together. I must be missing something.

Is there a video or 2 that I can watch?

TIA, Darryl Freinkel.

**Comment by a4ginatl** [comment]
Does Python use the VSC connections to the IBM i?
Darryl.

**Comment by a4ginatl** [comment]
Do I need to set up a shared folder on the IBM i?
Darryl.

**Comment by Wright4i** [comment, marked as answer]
Hi Darryl,

Code for IBM i and Python on the IBM i are unrelated and no Python doesn't use Visual Studio Code to connect to the IBM i.

To execute Python on the IBM i you need to place your `.py` file onto the IFS and open a terminal (BASH/PuTTY/QP2TERM) and execute it using `python3 example.py`

Here are some resources:
https://ibm.github.io/ibmi-oss-resources/
https://www.omniuser.org/downloads/omni202104MikePavlak_1-IntrotoPython-6.3.pdf 

For the most part other than connecting to DB2 (using a package like `pyodbc` or `ibm_db_dbi`) using Python on the IBM i is the same as developing Python on Windows or Linux. You can run Python on your local machine, develop, then promote to the IBM i after testing your code.


I am going to close this because it's unrelated to Code for IBM i. But please check out that first link for IBM i OSS Resources there is a community link there called Ryver where you can sign up and talk about Python on the IBM i and get a lot more help!
